python/ray/dashboard/modules/event/event_agent.py

```diff
@@ -47,33 +47,27 @@ def __init__(self, dashboard_agent):
 
         logger.info("Event agent cache buffer size: %s", self._cached_events.maxsize)
 
-    async def _connect_to_dashboard(self):
-        """Connect to the dashboard. If the dashboard is not started, then
-        this method will never returns.
-
-        Returns:
-            The ReportEventServiceStub object.
+    async def _get_dashboard_http_address(self):
+        """
+        Lazily get the dashboard http address from InternalKV. If it's not set, sleep
+        and retry forever.
         """
         while True:
+            if self._dashboard_http_address:
+                return self._dashboard_http_address
             try:
-                dashboard_rpc_address = await self._gcs_aio_client.internal_kv_get(
-                    dashboard_consts.DASHBOARD_RPC_ADDRESS.encode(),
+                dashboard_http_address = await self._gcs_aio_client.internal_kv_get(
+                    ray_constants.DASHBOARD_ADDRESS.encode(),
                     namespace=ray_constants.KV_NAMESPACE_DASHBOARD,
-                    timeout=1,
+                    timeout=dashboard_consts.GCS_RPC_TIMEOUT_SECONDS,
                 )
-                dashboard_rpc_address = dashboard_rpc_address.decode()
-                if dashboard_rpc_address:
-                    logger.info("Report events to %s", dashboard_rpc_address)
-                    options = ray_constants.GLOBAL_GRPC_OPTIONS
-                    channel = utils.init_grpc_channel(
-                        dashboard_rpc_address, options=options, asynchronous=True
-                    )
-                    return event_pb2_grpc.ReportEventServiceStub(channel)
+                if not dashboard_http_address:
+                    raise ValueError("Dashboard http address not found in InternalKV.")
+                self._dashboard_http_address = dashboard_http_address.decode()
+                return self._dashboard_http_address
             except Exception:
-                logger.exception("Connect to dashboard failed.")
-            await asyncio.sleep(
-                event_consts.RETRY_CONNECT_TO_DASHBOARD_INTERVAL_SECONDS
-            )
+                logger.exception("Get dashboard http address failed.")
+            await asyncio.sleep(1)
```

Comment:
We should remove `event_consts.RETRY_CONNECT_TO_DASHBOARD_INTERVAL_SECONDS` since its dead now